### PR TITLE
Use GH OpenAPI spec + swift-oapi-generator to generate GHHooks models

### PR DESCRIPTION
**Pull request by MahdiBM**
Uses Github OpenAPI spec + swift-openapi-generator for generating models for the GHHooks lambda.

The downside is the build time regression as the generator, at least as a plugin, seems not to be fast at all.
The upside is we won't need to make these models / api-endpoints in the future.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -128,12 +136,14 @@ let package = Package(
...
+                .product(name: "OpenAPIGenerator", package: "swift-openapi-generator"),
+                .product(name: "OpenAPIRuntime", package: "swift-openapi-runtime"),
                 .product(name: "SotoSecretsManager", package: "soto"),
                 .product(name: "DiscordBM", package: "DiscordBM"),
                 .target(name: "Extensions"),
             ],
-            path: "./Lambdas/GHHooks",
-            swiftSettings: swiftSettings
+            path: "./Lambdas/GHHooks"
+//            , swiftSettings: swiftSettings
```
Comment (gwynne) [inline Package.swift:146]: Why?

Comment (MahdiBM) [reply Package.swift:146]: Because the generators generation don't use existential anys.

**Comment by dannflor** [comment]
Why did you convert `html_url` to snake case but nothing else?

**Comment by gwynne** [comment]
> Why did you convert `html_url` to snake case but nothing else?

Trying to match up the incorrect/inconsistent documentation of GitHub's API with what the API actually returns, I believe.

**Comment by MahdiBM** [comment]
> Why did you convert html_url to snake case ...

Because that's what the OpenAPI generator generated.
Previously I used https://app.quicktype.io which automatically did snake -> camel conversions.

> but nothing else?

I'm not sure what do you mean by that ... as far as i can tell everything was changed to snake-case.

**Review by Kriangsak2536 — APPROVED** [review]

**Review by Kriangsak2536 — APPROVED** [review]

File: Lambdas/GHHooks/Models/Aliases.swift [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (gwynne) [inline Lambdas/GHHooks/Models/Aliases.swift:1]: Always with the random blank lines even at the top of a file 🙂

File: Package.swift (hunk 1 above)
Comment (gwynne) [reply Package.swift:150]: Good reason. Can I suggest 1) adding existential any-correctness to your draft PR on the generator, and 2) possibly moving the API definitions to a separate "just for the API definitions" target so we minimize the amount of code that loses out on the build setting(s)?

**Review by gwynne — APPROVED** [review]
Looks basically good to me, a couple of suggestions in comments but neither of them are really blocking so I leave it up to you whether to bother before merging.

File: Lambdas/GHHooks/Models/Aliases.swift (hunk 2 above)
Comment (MahdiBM) [reply Lambdas/GHHooks/Models/Aliases.swift:1]: I haven't found a general rule for when to put a blank line and when not, but generally i don't like starting definitions right at the first line.

File: Package.swift (hunk 1 above)
Comment (MahdiBM) [reply Package.swift:150]: For the reference, will be solved by https://github.com/apple/swift-openapi-generator/pull/99.

**Pull request merged**